libs/base/ui/graphical/icon/src/icon.component.ts

```diff
@@ -33,17 +31,20 @@ export class IconComponent extends LitElement implements IconProperties {
 
   render(): TemplateResult {
     return html`
-      <slot>
-        ${when(this.type, () =>
-          this.renderer
-            ? this.renderer
-            : svg`
-                <svg viewBox="0 0 24 24">
-                  <use href="${this.spriteUrl}" />
-                </svg>
-              `
-        )}
-      </slot>
+      ${when(
+        this.renderer,
+        () => this.renderer,
+        () =>
+          when(
+            this.spriteUrl,
+            () => svg`
+              <svg viewBox="0 0 24 24">
+                <use href="${this.spriteUrl}" />
+              </svg>
+            `,
+            () => html`<slot></slot>`
+          )
+      )}
```

Comment:
I'd use standard JS here, and not wire it into the template logic. 

Also: `protected override render(): TemplateResult | void {`